### ISSUE TITLE
Function 'PowBackward0' returned nan values in its 0th output.

### ISSUE THREAD

**Issue by marmikp** [dreamquark-ai/tabnet#232]
Hello,

I am using the **code of self-supervised learning** for my own data. My dataset contains some columns which values are the same for all the samples, therefore the standard deviation is zero for those columns during calculating the loss and this is the reason for occurring this error. I can't remove those columns because in the future I'll have some different values rather than the same value. The same issue is occurring in the forest cover type dataset.

Can you please suggest me to which method I should apply for those columns?
Should I use the MSELoss instead of TabNet's proposed loss function?

Thank you

**Comment by Optimox** [comment]
Hello @marmikp,

Have you been working from the branch of this PR for self supervised learning: #220 ?

It looks like a weird case to have a column with only one value. Note that pretraining that way probably won't help you much after because the model will have learn to predict the unique value. If you do not have access to rows with other values, maybe you could artificially change some values so that they are not always the same.

Could you also share the error message you get please?
